home work 23 Form/src/js/validator.js

```diff
@@ -0,0 +1,93 @@
+import ValidationError from "./validation-error.js";
+
+export const Validator = {
+  errors: {},
+  validators: {
+    isNotEmpty: {
+      validate: (value) => value !== "",
+      message: "The field can't be empty",
+      errorType: "required",
+    },
+    isNumber: {
+      validate: (value) => !isNaN(value) && Number.isInteger(+value),
+      message: "The field should has a integer",
+      errorType: "number",
+    },
+    length(min, max) {
+      return {
+        validate: (value) => value.length <= max && value.length >= min,
+        message: `Username must be between ${min} and ${max} characters`,
+        errorType: "maxLength",
+      };
+    },
+    isEmail() {
+      return {
+        validate: (value) => {
+          const emailRegex =
+            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+          return emailRegex.test(value);
+        },
+        message: "Invalid email format",
+        errorType: "email",
+      };
+    },
+    isValidPassword: {
+      validate: (value) =>
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*])\S{8,}$/.test(value),
+      message:
+        "Password must have at least 8 characters that include at least 1 lowercase character, 1 uppercase character, 1 number, and 1 special character in (!@#$%^&*)",
+      errorType: "password",
+    },
+    isPasswordMatch: {
```

Comment:
confirm password is red when passwords are matched 
<img width="305" alt="image" src="https://user-images.githubusercontent.com/24583373/221429900-34719ee6-ef73-4e2e-8d8d-912068c62ff5.png">

confirm password is green when passwords ain't matched 

<img width="314" alt="image" src="https://user-images.githubusercontent.com/24583373/221429971-c343036b-2da3-4f97-9196-7beab2a6b66b.png">